controlsfx/src/main/java/impl/org/controlsfx/skin/NotificationPaneSkin.java

```diff
@@ -112,20 +113,27 @@ public NotificationPaneSkin(final NotificationPane control) {
             }
         });
         registerChangeListener(control.closeButtonVisibleProperty(), e -> notificationBar.updatePane());
+        
+        getSkinnable().focusedProperty().addListener((o, ov, hasFocus) -> {
+            if (hasFocus) {
+                if (getChildren().size() > 0) {
+                    getChildren().get(0).requestFocus();
+                }
+            }
+        });
 
-        // Fix for Issue #522: Prevent NotificationPane from receiving focus
-        ParentTraversalEngine engine = new ParentTraversalEngine(getSkinnable());
-        ReflectionUtils.setTraversalEngine(control, engine);
-        engine.setOverriddenFocusTraversability(false);

```

Comment:
Adding a fix for these 3 lines has made me realized the importance of a Traversal Engine in JavaFX. Most of the heavy lifting was already done by Jonathan in [JDK-8091673](https://bugs.openjdk.java.net/browse/JDK-8091673). @Maxoudela @eugener Do you think its worth pursuing this?

Reply:
AFAIU, it disables the Parent nodes in a custom control from receiving focus. It was implemented as a fix for https://github.com/controlsfx/controlsfx/issues/713  by @effad

Reply:
I can confirm what @abhinayagarwal has said about these lines. The issue-number #522 refers to BitBucket-Issues (no longer existant ...).
I also would appreciate if the TraversalEngine would be part of the public API. After all, the above lines are very hacky :-).